x-pack/plugins/infra/public/pages/metrics/hosts/components/search_bar/controls_title.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiFormLabel, EuiPopover, EuiIcon, EuiToolTip, EuiLink } from '@elastic/eui';
+import useToggle from 'react-use/lib/useToggle';
+import { css } from '@emotion/react';
+import { i18n } from '@kbn/i18n';
+import { availableControlsPanels } from '../../hooks/use_control_panels_url_state';
+
+const helpMessages = {
+  [availableControlsPanels.SERVICE_NAME]: {
+    text: `${i18n.translate('xpack.infra.hostsViewPage.serviceNameControl.popoverHelpLabel', {
+      defaultMessage: 'Services detected via',
+    })}`,
+    link: {
+      text: `${i18n.translate('xpack.infra.hostsViewPage.serviceNameControl.popoverHelpLink', {
+        defaultMessage: 'APM',
+      })}`,
+      href: 'https://ela.st/docs-infra-apm',
+      ['data-test-subj']: 'hostsViewServiceNameControlPopoverHelpLink',
+    },
+  },
+};
```

Comment:
I mean, considering that `helpMessage` can now be just `ReactNode`, do we still need to map those help messages to `embeddableId`? How I see it, `<ControlTitle>` just looks at whether `helpMessage` is there and renders whatever is inside without any knowledge about how those messages are generated. Let me know if you want to chat about this, not sure how clear my explanations is.

Reply:
The idea is that [here](https://github.com/elastic/kibana/pull/172094/files#diff-3728878ef364d122845f374c13597cdd90fab8e453118553837ce1e5da983476R63-R71) we get all controls and based on the id we match the messages - do you mean that we want to have the message and match the id there and pass the message itself to the component? I am trying to understand what you mean and what's the benefit because we want to render the message for the specific control ( I imagine in the future we can have more controls with this functionality so `helpMessages` will have more than one message)

Reply:
> do you mean that we want to have the message and match the id there and pass the message itself to the component?

Yes, exactly, to avoid passing embeddable id into the `<ControlTitle>` I suggest passing the help message directly. It seems more natural to me that `controls_content.tsx` would hold the knowledge about multiple help messages because it responsible for rendering the `ControlGroup` with multiple components and `<ControlTitle>` would be responsible for a single specific title without the knowledge that is part of a bigger control group.

I feel like I'm nitpicking at this point tbh 🙈 😅 Feel free to ignore this if it does not make sense, I might be missing a bigger context here.

Reply:
I see what you mean now - we can move the `helpMessages` to the control content but the embeddable id is still something we need to pass because of the [form label](https://github.com/elastic/kibana/pull/172094/files#diff-775f472b1b89f1582708c8049e465507c537b5d1ada7ec3798b08d6ed27d6f88R63). We can still do the messages matching outside of the component but we won't get rid of the embeddable prop unfortunately so it will just add an extra property passed so that's why I kept it there (also this title component is useful only in this context so I don't see it being reused as a component outside of the controls).

Reply:
Thank you, I totally missed the form label thing. You're right, there is no point in having both properties if we cannot get rid of embaddable id